pkg/kubelet/kubelet_test.go

```diff
@@ -2265,6 +2265,92 @@ func TestMakeEnvironmentVariables(t *testing.T) {
 				{Name: "POD_NAMESPACE", Value: "downward-api"},
 			},
 		},
+		{
+			name: "env expansion",
+			ns:   "test1",
+			container: &api.Container{
+				Env: []api.EnvVar{
```

Comment:
Yeah - I also need to have tests that establish that the ordering semantics for resolution are correctly implemented.